### PR TITLE
Add telemetry for breadcrumbs and have the setting apply without refresh

### PR DESCRIPTION
**Pull request by turt2live**
Fixes https://github.com/vector-im/riot-web/issues/9207

### CODE REVIEW

File: src/Analytics.js [hunk 1]
```diff
@@ -234,6 +239,11 @@ class Analytics {
         this._setVisitVariable('RTE: Uses Richtext Mode', state ? 'on' : 'off');
     }
 
+    setBreadcrumbs(state) {
+        if (this.disabled) return;
+        this._setVisitVariable('Breadcrumbs', state ? 'enabled' : 'disabled');
```
Comment (bwindels) [inline src/Analytics.js]: Is this safe to call repeatedly from componentDidUpdate? It pushes to an array...

Comment (turt2live) [reply src/Analytics.js]: I believe so. It didn't blow up in testing, and matomo is built for much busier applications than us.

Comment (bwindels) [reply src/Analytics.js]: Let me rephrase that: is this going to send telemetry every time LeftPanel's `componentDidUpdate` method is called? If so, this doesn't seem like the right event, I'd assume we want to send telemetry upon a breadcrumbs related user interaction?

Comment (turt2live) [reply src/Analytics.js]: Oh... Yup, we want to know when it changes not every time something happens. Will fix.

Comment (turt2live) [reply src/Analytics.js]: turns out that the arguments to didUpdate are also the old state, not the new state: https://reactjs.org/docs/react-component.html#componentdidupdate

File: src/components/structures/LeftPanel.js [hunk 2]
```diff
@@ -73,6 +86,22 @@ const LeftPanel = React.createClass({
...
+    componentDidUpdate(prevProps, prevState) {
+        if (prevState.breadcrumbs !== this.state.breadcrumbs) {
+            Analytics.setBreadcrumbs(this.state.breadcrumbs);
+        }
+    },
+
+    _onBreadcrumbsChanged: function(settingName, roomId, level, valueAtLevel, value) {
+        // Features are only possible at a single level, so we can get away with using valueAtLevel.
+        // The SettingsStore runs on the same tick as the update, so `value` will be wrong.
+        this.setState({breadcrumbs: valueAtLevel});
```
Comment (bwindels) [inline src/components/structures/LeftPanel.js]: could also send analytics here, but componentDidUpdate will also work.

Comment (turt2live) [reply src/components/structures/LeftPanel.js]: Theory was that componentDidUpdate would catch all the places it could be updated, but with a single place it could be updated at it might not make sense... I'll leave it in componentDidUpdate for now and consider moving it when the labs flag gets replaced with a real setting.
